### PR TITLE
Airqo PM 2.5 calibrated values

### PR DESCRIPTION
**Pull request by NoahNsimbe**
# Calibrated Values

**_WHAT DOES THIS PR DO?_**
Adds the calibrated pm 2.5 value for AirQo device measurements

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
None
**_WHAT IS THE LINK TO THE PR BRANCH_**
[ft-calibrate-values](https://github.com/airqo-platform/AirQo-api/tree/ft-calibrate-values)
**_HOW DO I TEST OUT THIS PR?_**
checkout this branch [ft-calibrate-values](https://github.com/airqo-platform/AirQo-api/tree/ft-calibrate-values)
```
cd AirQo-api/src/data-mgt/python/jobs
python3 -m venv venv
source venv/bin/activate
export CALIBRATE_URL=<CALIBRATE_URL>
export FEEDS_BASE_URL=<FEEDS_BASE_URL>
export DEVICE_REGISTRY_STAGING_URL=<DEVICE_REGISTRY_STAGING_URL>
export DEVICE_REGISTRY_PRODUCTION_URL=<DEVICE_REGISTRY_PRODUCTION_URL>
pip install -r requirements.txt
python airqo_device_measurements.py
```
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
None
**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
None
**_ARE THERE ANY RELATED PRs?_**
None

### CODE REVIEW

**Review by Priscilla-AL — COMMENTED** [review]
Thanks, Noah, where are these values stored? I can't find them in Mongo

**Comment by NoahNsimbe** [comment]
> Thanks, Noah, where are these values stored? I can't find them in Mongo

It is easier to locate them using the endpoint

**Comment by bobturneruk** [comment]
Does this use the calibrate microservice to periodically create and store calibrated values in mongodb, please?

**Comment by NoahNsimbe** [comment]
> Does this use the calibrate microservice to periodically create and store calibrated values in mongodb, please?

Yes, it uses calibrate microservice to create calibrated values before storing in mongodb…that's all. Storing in mongodb is done by device registry microservice

**Review by ja-odur — APPROVED** [review]
LGTM, Although when run, it seems to overload the mongo db.

**Comment by NoahNsimbe** [comment]
> LGTM, Although when run, it seems to overload the mongo db.

Yes, I also realized it.

**Review by Baalmart — COMMENTED** [review]
Thanks @NoahNsimbe All runs well, just curious about a few things:

- How do we switch environments? Or is it storing in 2 environments at the same time?
- Getting some bad request messages from the calibrate endpoint logs, something along the lines of: "`Please specify the device_id, datetime, pm2.5, pm10, temperature and humidity values in the body. Refer to the API documentation for details.`".
- Where are the deployment yaml files situated in this api codebase?

File: src/data-mgt/python/jobs/events.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 DEVICE_REGISTRY_STAGING_URL = os.getenv("DEVICE_REGISTRY_STAGING_URL")
 DEVICE_REGISTRY_PRODUCTION_URL = os.getenv("DEVICE_REGISTRY_PRODUCTION_URL")
+ENVIRONMENT = os.getenv("ENVIRONMENT", "STAGING")
```
Comment (Baalmart) [inline src/data-mgt/python/jobs/events.py:8]: Is this where we switch environments during tests?
Interested to learn about how the deployments for the different ENVs are handled.

Comment (NoahNsimbe) [reply src/data-mgt/python/jobs/events.py:8]: @Baalmart, there is no switching of environments, if the `ENVIRONMENT` variable is set to `PRODUCTION`, the measurements are sent to both staging and production else they are sent to staging.

**Comment by NoahNsimbe** [comment]
> Thanks @NoahNsimbe All runs well, just curious about a few things:
> 
> * How do we switch environments? Or is it storing in 2 environments at the same time?
> * Getting some bad request messages from the calibrate endpoint logs, something along the lines of: "`Please specify the device_id, datetime, pm2.5, pm10, temperature and humidity values in the body. Refer to the API documentation for details.`".
> * Where are the deployment yaml files situated in this api codebase?

@Baalmart, this is a cloud function, the command for  deployment is in the Readme.md... Going to share the `.env.yaml` file on slack

**Comment by Baalmart** [comment]
> > Thanks @NoahNsimbe All runs well, just curious about a few things:
> > 
> > * How do we switch environments? Or is it storing in 2 environments at the same time?
> > * Getting some bad request messages from the calibrate endpoint logs, something along the lines of: "`Please specify the device_id, datetime, pm2.5, pm10, temperature and humidity values in the body. Refer to the API documentation for details.`".
> > * Where are the deployment yaml files situated in this api codebase?
> 
> @Baalmart `device_id`, `datetime`, `pm2.5`, `pm10`, `temperature` and `humidity` must be provided for calibration, if any of them is missing, then you get that error. Note that attimes temperature and humidity are not present in the `feeds` response

Cool, makes sense. Thanks.

**Review by Baalmart — APPROVED** [review]
LGTM

**Review by danielogen — APPROVED** [review]
LGTM

**Pull request merged**